recipes-devtools/rust/rust.inc

```diff
@@ -192,8 +217,27 @@ def llvm_cpu(d):
     except:
         return trans.get(target, "generic")
 
-TARGET_LLVM_CPU="${@llvm_cpu(d)}"
-TARGET_LLVM_FEATURES = "${@llvm_features_from_tune(d)} ${@llvm_features_from_cc_arch(d)}"
+# generates our target CPU value for cross builds
+def llvm_cross_cpu(d):
+    feat = d.getVar('TUNE_FEATURES', True)
+    if not feat:
+        return ""
+    feat = frozenset(feat.split())
+
+    # The only reliable way of finding the cpu is to look in TUNE_FEATURES
+    # which is derived from DEFAULTTUNE
+    if "mips32r2" in feat:
+        return "mips32r2"
+    elif "mips32" in feat:
+        return "mips32"
+
+    # TODO: Proper x86 targets. "generic" is a valid choice.
+    # TODO: Proper ARM cpu selection, for now "generic" is a valid cpu choice.
+    return "generic"
+
+TARGET_LLVM_CPU="${@llvm_cross_cpu(d)}"
```

Comment:
Hmm, I'll take a look at the `PACKAGE_ARCH` bits in detail again when I have a chance, I had thought it did what we wanted here. That said, it does look like it's getting set to BUILD_ARCH by cross.bbclass, which is almost certainly not what we want.

Note that the rust-cross targets don't actually build anything at this point, they're only cross in the sense that they drop files in the native sysroot but need some details about the target to generate them.

Solutions here are a bit tricky due to the way json files are generated. Likely could be made better by only generating 1 per recipe (instead of trying to do all of HOST/BUILD/TARGET).